test/user.test.js

```diff
@@ -858,6 +858,31 @@ describe('User', function() {
         }, done);
       });
 
+      it('Should report 302 when redirect url is set', function(done) {
+        testConfirm(function(result, done) {
+          request(app)
+            .get('/users/confirm?uid=' + (result.uid) +
+              '&token=' + encodeURIComponent(result.token) +
+              '&redirect=' + encodeURIComponent(options.redirect))
+            .expect(302);
+
+          done();
```

Comment:
You have to pass `done` as a callback of `expect` and/or `end`.

The current implementation marks the test as done before the HTTP request is finished.

The same applies to the next test below.

```js
.expect(302)
.end(done);
```